### PR TITLE
Added test for WKO: Add 'domain.spec.maxClusterUnavailable' (as default for 4.0 'cluster.spec.maxUnavailable')

### PR DESCRIPTION
**Pull request by hzhao-github**
test cases:

Start a domain with 2 clusters, cluster-1 and cluster-2

    Test domain.spec.maxClusterUnavailable and domain.spec.replicas

    Config Config domain.spec.maxClusterUnavailable = 2 and Config domain.spec.replicas = 4 in domain resource
    During rolling restart, verify that the Operator rolling restarts 2 managed server concurrently until all 4 managed servers are rolling restarted

    Test cluster.spec.maxUnavailable and cluster.spec.replicas

    Config Config domain.spec.maxClusterUnavailable =2 and Config domain.spec.replicas = 4 in domain resource
    Config maxUnavailable =1 and replicas = 2 in cluster-1 resource
    During rolling restart
    ¿ verify that the Operator rolling-restarts 2 managed server in cluster-1one by one, not concurrently
    ¿ verify that the Operator rolling-restarts 2 managed server in cluster-2 concurrently until all 4 managed servers are rolling restarted

    Test domain.spec.maxClusterConcurrentShutdown

    Config domain.spec.replicas = 5 to startup 5 managed servers in two clusters
    Config domain.spec.maxClusterConcurrentShutdown = 2
    Config domain.spec.replicas = 1 to scale down the clusters, verify that the Operator shuts down 2 managed servers concurrently in each cluster until 4 managed servers are shutdown

    Test cluster.spec.maxConcurrentShutdown

    Config domain.spec.replicas = 5 to startup 5 managed servers in clusters
    Config domain.spec.maxClusterConcurrentShutdown = 2
    Config cluster.spec.maxConcurrentShutdown = 3
    Config cluster.spec.replicas = 2 in cluster-1 resource to scale down cluster-1, verify that the Operator shuts down 3 managed servers concurrently in cluster-1
    Config domain.spec.replicas = 3 in domain resource to scale down cluster-2, verify that the Operator shuts down 2 managed servers concurrently in cluster-2

    Test domain.spec.maxClusterConcurrentStartup

    Config domain.spec.replicas = 1 to startup 1 managed servers in two clusters
    Config domain.spec.maxClusterConcurrentStartup = 2
    Config domain.spec.replicas = 3 to scale up the clusters, verify that the Operator startup 2 managed servers concurrently in each cluster

    Test cluster.spec.maxConcurrentStartup

    Config domain.spec.replicas = 1 to startup 1 managed servers in two clusters
    Config domain.spec.maxClusterConcurrentStartup = 2
    Config cluster.spec.maxConcurrentStartup= 2 in cluster-1
    Config cluster.spec.replicas = 3 in cluster-1 resource to scale up cluster-1, verify that the Operator starts up 2 managed servers concurrently
    Config cluster.spec.maxConcurrentStartup= 1 in cluster-2
    Config cluster.spec.replicas = 3 in cluster-2 resource to scale up cluster-2, verify that the Operator starts up 2 managed servers one by one, not concurrently.

Jenkins:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/14972/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/15102
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/15103

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java [hunk 1]
```diff
@@ -0,0 +1,760 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
```
Comment (sankarpn) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java:1]:
```suggestion
// Copyright (c) 2023, Oracle and/or its affiliates.
```

**Review by sankarpn — APPROVED** [review]
minor correction, otherwise looks good.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java (hunk 1 above)
Comment (rjeberhard) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java:1]: You need to update the copyright date on the other two files also.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java [hunk 2]
```diff
@@ -0,0 +1,760 @@
...
+ * 5. cluster.spec.maxConcurrentShutdown
+ * 6. cluster.spec.maxConcurrentStartup.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("Test to a create model in image domain with Cluster Resourcees")
+@IntegrationTest
+@Tag("olcne")
+@Tag("oke-parallel")
+@Tag("kind-parallel")
+@Tag("okd-wls-srg")
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java:94]: You need to verify the test result in OKE, OKD and olcne before adding the tags.

Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMaxConcurOptions.java]: removed 

@Tag("olcne")
@Tag("oke-parallel")
@Tag("okd-wls-srg")

**Review by xiancao — APPROVED** [review]

**Pull request merged**